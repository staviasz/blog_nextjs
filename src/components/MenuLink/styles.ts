import styled, { css } from 'styled-components';

export const MenuLink = styled.a`
  ${({ theme }) => css`
    display: block;
    color: ${theme.colors.white};
    text-decoration: none;
    margin-bottom: ${theme.spacings.small};
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 150ms ease-in-out;

    &:hover {
      border-right-color: ${theme.colors.secondary};
      color: ${theme.colors.secondary};
    }
  `}
`;

export const MenuDiv = styled.div`
  ${({ theme }) => css`
    display: block;
    color: ${theme.colors.white};
    text-decoration: none;
    margin-bottom: ${theme.spacings.small};
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 150ms ease-in-out;

    &:hover {
      border-right-color: ${theme.colors.secondary};
      color: ${theme.colors.secondary};
    }
  `}
`;
