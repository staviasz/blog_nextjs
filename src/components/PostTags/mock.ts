import { data } from '../../api/data.json';

const { posts } = data;
export default posts;
