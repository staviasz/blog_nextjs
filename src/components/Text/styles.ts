import styled, { css } from 'styled-components';

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxsmall};
    text-align: left;

    p {
      margin: ${theme.spacings.xlarge} 0;
    }

    a {
      color: ${theme.colors.secondary};
    }

    a:hover {
      filter: brightness(50%);
    }

    pre {
      background-color: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
    }

    img {
      max-width: 100%;
      padding: ${theme.spacings.xxlarge} 0;
    }
  `}
`;
