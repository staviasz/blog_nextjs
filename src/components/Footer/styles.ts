import styled, { css } from 'styled-components';
import { Text } from '../Text/styles';

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    ${Text} {
      font-size: ${theme.font.sizes.small};
      text-align: center;
      margin: 0;
    }
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
