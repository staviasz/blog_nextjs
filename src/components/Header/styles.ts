import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';
import { LinkContent } from '../LogoLink/styles';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.xhuge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;
    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }
    ${LinkContent} {
      > img {
        max-height: 100%;
        width: 12rem;
        height: 12rem;
        border-radius: 50%;
      }
    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
