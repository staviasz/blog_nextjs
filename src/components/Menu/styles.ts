import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  $menuvisible: string;
};

const ContainerChanger = (
  menuvisible: MenuBehaviorProps['$menuvisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuvisible === 'open' ? '0' : '-30rem'};
  overflow-y: ${menuvisible === 'open' ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${menuvisible === 'open' ? '0' : '-32rem'};
  }
`;

export const Container = styled.div<MenuBehaviorProps>`
  ${({ theme, $menuvisible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;
    ${ContainerChanger($menuvisible, theme)}

    img {
      width: calc(${theme.spacings.xhuge} * 3);
      height: calc(${theme.spacings.xhuge} * 3);
      border-radius: 50%;
    }
  `}
`;

export const Nav = styled.nav`
  margin: auto;
  width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xxlarge};

      img {
        border: 0.5rem solid ${theme.colors.secondary};
      }
    }
  `}
`;

const buttonChanger = (
  menuvisible: MenuBehaviorProps['$menuvisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuvisible ? '26rem' : '1rem'};
  color: ${menuvisible ? theme.colors.secondary : theme.colors.white};

  @media ${theme.media.lteMedium} {
    left: ${menuvisible ? '26rem' : '-0.5rem'};
  }
`;

export const OpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, $menuvisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    transition: all 300ms ease-in-out;
    ${buttonChanger($menuvisible, theme)}
  `}
`;
