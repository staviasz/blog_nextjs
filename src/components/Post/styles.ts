import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface PostContainerProps {
  size: 'max' | 'content';
}

const postContainerStyles = {
  max: (theme: Theme) => css`
    max-width: ${theme.sizes.max};
  `,
  content: (theme: Theme) => css`
    max-width: ${theme.sizes.content};
  `,
};

export const PostContainer = styled.article<PostContainerProps>`
  ${({ theme, size }) => css`
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    ${postContainerStyles[size](theme)}

    @media ${theme.media.lteMedium} {
      padding: 0 calc(${theme.spacings.large} / 2);
    }
  `}
`;

export const Content = styled.div``;
