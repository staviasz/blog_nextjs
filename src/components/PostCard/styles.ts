import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';
import { Title as HeadingStyles } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: calc(${theme.spacings.small} - 0.6rem) 0;
      /* margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small}; */
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    &:hover a {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
    }
  `}
`;

export const Cover = styled.img`
  max-width: calc(${theme.spacings.xlarge} * 10);
  width: 100%;
  max-height: calc(${theme.spacings.xxsmall} * 11);
  height: 100%;
  transition: opacity 300ms ease-in-out;
  border-radius: 4px;
  border: 1px solid #ddd;
`;

export const Excerpt = styled.p`
  margin: 0;
`;
