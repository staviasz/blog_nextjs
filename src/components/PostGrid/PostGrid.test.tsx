import { screen } from '@testing-library/react';
import { PostGrid, PostGridProps } from '.';
import { renderTheme } from '../../styles/render-theme';

import { mapPosts } from '../../utils/mapData/mapPost';
import mock from './mock';

const postsData = mapPosts(mock);
const props: PostGridProps = { posts: postsData };

describe('<PostGrid />', () => {
  it('should not render posts', () => {
    renderTheme(<PostGrid {...props} posts={undefined} />);

    expect(screen.getByText(/Nenhum post/i)).toBeInTheDocument();
  });

  it('should render three posts', () => {
    const { container } = renderTheme(<PostGrid posts={props.posts} />);

    expect(screen.queryByText(/Nenhum post/i)).not.toBeInTheDocument();

    expect(screen.getAllByRole('heading')).toHaveLength(2);
    expect(screen.getAllByRole('img')).toHaveLength(1);
    expect(container.querySelectorAll('p')).toHaveLength(2);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostGrid posts={props.posts} />);
    expect(container).toMatchSnapshot();
  });
});
