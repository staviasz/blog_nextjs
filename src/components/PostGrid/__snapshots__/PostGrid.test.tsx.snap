// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> should match snapshot 1`] = `
.c4 {
  font-size: 2.4rem;
  text-transform: capitalize;
  font-style: italic;
}

.c2 .c3 {
  margin: calc(1.4rem - 0.6rem) 0;
}

.c2 a {
  text-decoration: none;
  color: inherit;
  transition: all 300ms ease-in-out;
}

.c2:hover a {
  color: #dc143c;
}

.c2:hover img {
  opacity: 0.8;
}

.c6 {
  max-width: calc(4.0rem * 10);
  width: 100%;
  max-height: calc(1.8rem * 11);
  height: 100%;
  transition: opacity 300ms ease-in-out;
  border-radius: 4px;
  border: 1px solid #ddd;
}

.c5 {
  margin: 0;
}

.c0 {
  width: 100%;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

@media (max-width: 500px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          <a
            href="/senioridade"
          >
            Junior, pleno e senior
          </a>
        </h2>
        <p
          class="c5"
        >
          Senioridade em desenvolvimento de sofware junior, pleno e senior
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/desenvolvimento-web"
        >
          <img
            alt="desenvolvimento web"
            class="c6"
            src="https://res.cloudinary.com/drko8mcj9/image/upload/v1691449461/photo_1498050108023_c5249f4df085_df84325049.webp"
          />
        </a>
        <h2
          class="c3 c4"
        >
          <a
            href="/desenvolvimento-web"
          >
            desenvolvimento web
          </a>
        </h2>
        <p
          class="c5"
        >
          As varias areas de desenvolvimento web
        </p>
      </div>
    </div>
  </div>
   
</div>
`;
